bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcarePetHandler.java

```diff
@@ -0,0 +1,250 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SurePetcarePetHandler} is responsible for handling the things created to represent Sure Petcare pets.
+ *
+ * @author Rene Scherer - Initial Contribution
+ * @author Holger Eisold - Added pet feeder status, location time offset
+ */
+@NonNullByDefault
+public class SurePetcarePetHandler extends SurePetcareBaseObjectHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SurePetcarePetHandler.class);
+
+    private static final ByteArrayFileCache IMAGE_CACHE = new ByteArrayFileCache("org.openhab.binding.surepetcare");
+
+    public SurePetcarePetHandler(Thing thing, SurePetcareAPIHelper petcareAPI) {
+        super(thing, petcareAPI);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateThingCache.getValue();
+        } else {
+            switch (channelUID.getId()) {
+                case PET_CHANNEL_LOCATION:
+                    logger.debug("Received location update command: {}", command.toString());
+                    if (command instanceof StringType) {
+                        synchronized (petcareAPI) {
+                            SurePetcarePet pet = petcareAPI.getPet(thing.getUID().getId());
+                            if (pet != null) {
+                                String newLocationIdStr = ((StringType) command).toString();
+                                Integer newLocationId = Integer.valueOf(newLocationIdStr);
+                                // Only update if location has changed. (Needed for Group:Switch item)
+                                if ((pet.status.activity.where.equals(newLocationId)) || newLocationId.equals(0)) {
+                                    logger.debug("Location has not changed, skip pet id: {} with loc id: {}", pet.id,
+                                            newLocationId);
+                                } else {
+                                    try {
+                                        logger.debug("Received new location: {}", newLocationId);
+                                        petcareAPI.setPetLocation(pet, newLocationId, ZonedDateTime.now());
+                                        updateState(PET_CHANNEL_LOCATION,
+                                                new StringType(pet.status.activity.where.toString()));
+                                        updateState(PET_CHANNEL_LOCATION_CHANGED,
+                                                new DateTimeType(pet.status.activity.since));
+                                    } catch (NumberFormatException e) {
+                                        logger.warn("Invalid location id: {}, ignoring command", newLocationIdStr);
+                                    } catch (SurePetcareApiException e) {
+                                        logger.warn("Error from SurePetcare API. Can't update location {} for pet {}",
```

Comment:
You catch an exception but ignore the actual content of it.

Reply:
added throwable to logging.